javalin/src/main/java/io/javalin/config/FileUploadConfig.kt

```diff
@@ -0,0 +1,87 @@
+package io.javalin.config
+
+import io.javalin.http.util.MultipartUtil
+import jakarta.servlet.MultipartConfigElement
+
+/**
+ * This class contains the configuration for handling multipart file uploads
+ *
+ * @property cacheDirectory : the directory where files which exceed the maximum in memory size should be cached
+ * @property maxFileSize : the maximum size allowed (in bytes) for an individual uploaded file
+ * @property maxInMemoryFileSize : the maximum size allowed (in bytes) before uploads are cached to disk
+ * @property maxTotalRequestSize : the maximum size allowed (in bytes) for the entire multipart request
+ */
+class FileUploadConfig {
+    //@formatter:off
+    private var cacheDirectory = System.getProperty("java.io.tmpdir")
+    private var maxFileSize: Long = -1
+    private var maxInMemoryFileSize: Int = -1
+    private var maxTotalRequestSize: Long = -1
+    //@formatter:on
+
+    //when the configuration is initialized we override the preupload function in the multipart util to read the
+    //configuration from these settings rather than the current hard-coded method
+    init{
+        MultipartUtil.preUploadFunction = { req ->
+            if(req.getAttribute(MultipartUtil.MULTIPART_CONFIG_ATTRIBUTE) == null){
+                req.setAttribute(MultipartUtil.MULTIPART_CONFIG_ATTRIBUTE, multipartConfigElement())
+            }
+        }
+    }
+
+    /**
+     * This class represents the potential file size descriptors to avoid the use of hard-coded multipliers
+     */
+    enum class SizeUnit(val multiplier: Int){
```

Comment:
Can we make this a top level class?

Reply:
Yes, own class, not inner. I prefer if we still keep it in the same file.

Reply:
That's been resolved now.  I've pulled SizeUnit up to the top level but still within the same file